airflow/models/dag.py

```diff
@@ -451,6 +453,7 @@ def __init__(
             max_active_tasks = concurrency
         self._max_active_tasks = max_active_tasks
         self._pickle_id: int | None = None
+        self._display_name = dag_id if display_name is None else display_name
```

Comment:
I think this require some more thinking.
Ideally if we have display name for Dags set bu the user than dag_id should be automaticly generated by Airflow without user involvment - I think it should be a unique identifier that can not be changed.

Reply:
I think what Elad meant is the user should be able to just do something like

```python
DAG(name="俺のダッグ")
```

and _internally_ Airflow would automatically generate a `dag_id` for it (say with `uuid`). Since (ideally) all user-facing interface would just show the user-friendly name, the auto-generated ID can be basically anything and wouldn’t matter to the user at all. And if the user does care, they can always still pass in a custom `dag_id` explicitly:

```python
DAG(dag_id="my_dag", name="俺のダッグ")
```

Reply:
Yep. We discussed slugifying the name - that can also be an option, however I think adding optional name to be displayed next to id is a better approach - because having a unique id tha tis "known" and does not require to follow the very same slugifying algorithm is better (for any case where Human seeing a good name has to now what unique id is used)

But seems I am in a minority, and it's not something I feel strongly about, so I am ok with any solutiom - somebody hopefully finally will implement it rather than just talking about it (I think this is about 4th discussion about it I recall).